### PR TITLE
Update couple your code

### PR DESCRIPTION
**Pull request by fsimonis**
This PR updates the couple your code section.

Exceptions are:

* Advanced -> Dealing with moving meshes
* Advanced -> Time interpolation #257
* Porting guides

### CODE REVIEW

**Comment by fsimonis** [comment]
I must double-check, but I think `[...]` breaks code highlighting in C++ files.

Maybe we should go for:
```cpp
// ...

// or

/* ... */
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 1]
```diff
@@ -15,30 +15,36 @@ For volume coupling in 2D, mesh connectivity boils down to defining triangles an
...
-bool isMeshConnectivityRequired(int meshID);
+bool requiresMeshConnectivityFor(precice::string_view meshName);
 ```
 
 {% warning %}
 The API function `isMeshConnectivityRequired` is only supported since v2.3.
 {% endwarning %}
 
+{% warning %}
+The bulk API functions are only supported since v3.
```
Comment (IshaanDesai) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]:
```suggestion
The bulk API functions are only supported from v3.
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 2]
```diff
@@ -48,52 +54,45 @@ Quads are only supported since v2.1. For older version, the methods only exist a
...
+/* ... */
+
+// We define the unit square in 2D
+std::vector<double> coords{
+    0, 0, // A
+    0, 1, // B
+    1, 0, // C
+    1, 1  // D
+};
+std::vector<VertexID> vertexIDs(4);
```
Comment (IshaanDesai) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md:66]: Just curious: how does the user get access to the `VertexID` data type?

File: pages/docs/couple-your-code/couple-your-code-direct-access.md [hunk 3]
```diff
@@ -14,41 +14,48 @@ These API functions are work in progress, experimental, and are not yet released
...
+
+    // Get the spacial dimensionality of the mesh
+    const int dims = precice.getMeshDimensions(otherMesh);
+
+    // Allocate and fill the  'boundingBox' according to the interested region
+    // with the desired bounds, in our example we use the unit cube.
+    // Assuming dim == 3, means that the bounding box has dim * 2 == 6 elements.
+    std::vector<double> boundingBox {
+        0, 0, 0, // minimum corner
+        1, 1, 1, // maximum corner
```
Comment (IshaanDesai) [inline pages/docs/couple-your-code/couple-your-code-direct-access.md]:
```suggestion
        1, 1, 1 // maximum corner
```

File: pages/docs/couple-your-code/couple-your-code-direct-access.md [hunk 4]
```diff
@@ -14,41 +14,48 @@ These API functions are work in progress, experimental, and are not yet released
...
-    const int otherMeshSize = precice.getMeshVertexSize(otherMeshID);
+    const int otherMeshSize = precice.getMeshVertexSize(otherMesh);
 
-    // Now finally get the data. First allocate memory for the IDs and the
-    // vertices
-    std::vector<double> otherSolverVertices(otherMeshSize * dim);
-    std::vector<int>    ids(otherMeshSize);
+    // Now finally get information about the mesh vertices.
+    // First allocate memory for the IDs and coordinates
+    std::vector<double>   otherCoordinates(otherMeshSize * dim);
```
Comment (IshaanDesai) [inline pages/docs/couple-your-code/couple-your-code-direct-access.md]:
```suggestion
    std::vector<double> otherCoordinates(otherMeshSize * dim);
```

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1: just some minor comments.

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md (hunk 2 above)
Comment (fsimonis) [reply pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md:66]: It lives in `precice/Types.hpp`, which is included by the `Participant.hpp`.
Does that need an explanation in the example?

Comment (IshaanDesai) [reply pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md:66]: > Does that need an explanation in the example?

No, I think its fine if it is included by the `Participant.hpp`. Users will probably figure it out.

**Pull request merged**